### PR TITLE
Support specifying extraArgs for kubelet/scheduler/apiserver/controllermanager

### PR DESCRIPTION
**Pull request by hongtaiwang**
* support specifying schedulerExtraArgs by `;` delimited array `KUBE_SCHEDULER_EXTRA_ARGS` in cluster config file

* support specifying apiServerExtraArgs by `;` delimited array `APISERVER_EXTRA_ARGS` in cluster config file

* support specifying kubeControllerManagerExtraArgs by `;` delimited array `KUBE_CONTROLLER_MANAGER_EXTRA_ARGS` in cluster config file

* support specifying controlPlaneKubeletExtraArgs & workerKubeletExtraArgs by `;` delimited array `CONTROLPLANE_KUBELET_EXTRA_ARGS` or `WORKER_KUBELET_EXTRA_ARGS` in cluster config file

* fix audit log ytt bug when `ENABLE_AUDIT_LOGGING: 'true'`. extraArgs type is object instead of array

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

#### Special notes for your reviewer

tested in vsphere env with cusomize fake val:
```
```

cluster obj and kcp obj could be created successfully with the fake envs
```
spec:
  kubeadmConfigSpec:
    clusterConfiguration:
      apiServer:
        extraArgs:
          admission-control-config-file: /etc/kubernetes/kube-apiserver-admission-pss.yaml
          cloud-provider: external
          t1_1: val1_1
          t1_2: val1_2
          tls-cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
        extraVolumes:
        - hostPath: /etc/kubernetes/kube-apiserver-admission-pss.yaml
          mountPath: /etc/kubernetes/kube-apiserver-admission-pss.yaml
          name: admission-pss
          pathType: File
          readOnly: true
        timeoutForControlPlane: 8m0s
      controllerManager:
        extraArgs:
          cloud-provider: external
          t3_1: val3_1
          t3_2: val3_2
          tls-cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
      dns:
        imageRepository: projects-stg.registry.vmware.com/tkg
        imageTag: v1.8.6_vmware.12
      etcd:
        local:
          dataDir: /var/lib/etcd
          extraArgs:
            cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
          imageRepository: projects-stg.registry.vmware.com/tkg
          imageTag: v3.5.4_vmware.10
      imageRepository: projects-stg.registry.vmware.com/tkg
      networking: {}
      scheduler:
        extraArgs:
          t2_1: val2_1
          t2_2: val2_2
          tls-cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384

...

    initConfiguration:
      localAPIEndpoint: {}
      nodeRegistration:
        criSocket: /var/run/containerd/containerd.sock
        kubeletExtraArgs:
          cloud-provider: external
          node-labels: image-type=ova,os-name=photon,os-type=linux,run.tanzu.vmware.com/tkr=v1.24.6---vmware.1-tkg.1-zshippable
          t4_1: val4_1
          t4_2: val4_2
          tls-cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
        name: '{{ ds.meta_data.hostname }}'
    joinConfiguration:
      discovery: {}
      nodeRegistration:
        criSocket: /var/run/containerd/containerd.sock
        kubeletExtraArgs:
          cloud-provider: external
          node-labels: image-type=ova,os-name=photon,os-type=linux,run.tanzu.vmware.com/tkr=v1.24.6---vmware.1-tkg.1-zshippable
          t4_1: val4_1
          t4_2: val4_2
          tls-cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
```

mc created successfully with normal args

### CODE REVIEW

**Comment by wjun** [comment]
@randomvariable Does OCI clusterclass have the same requirement as other cloud providers to customize extraArgs here?

**Comment by hongtaiwang** [comment]
> @randomvariable Does OCI clusterclass have the same requirement as other cloud providers to customize extraArgs here?

hi @wjun , asked @DanielXiao about this question. He said oci clusterclass is not under our scope and no need to maintain this part.

File: providers/config_default.yaml [hunk 1]
```diff
@@ -71,6 +71,31 @@ BUILD_EDITION:
...
+KUBE_SCHEDULER_EXTRA_ARGS:
+
+#! KUBE_CONTROLLER_MANAGER_EXTRA_ARGS specified kube controller manager flags.
+#! It should be of the format "key1=value1;key2=value2".
+#! E.g. "tls-min-version=VersionTLS12;profiling=false"
+KUBE_CONTROLLER_MANAGER_EXTRA_ARGS:
+
+#! CONTROLPLANE_KUBELET_EXTRA_ARGS specified kubelet flags for control plane nodes.
+#! It should be of the format "key1=value1;key2=value2".
+#! E.g. "heartbeat-interval=300;election-timeout=2000"
```
Comment (DanielXiao) [inline providers/config_default.yaml:91]: These are etcd flags

Comment (hongtaiwang) [reply providers/config_default.yaml]: updated

**Review by DanielXiao — APPROVED** [review]

**Pull request merged**